src/main/java/com/aws/greengrass/logmanager/LogManagerService.java

```diff
@@ -504,68 +505,58 @@ private void handleCloudWatchAttemptStatus(CloudWatchAttempt cloudWatchAttempt)
         isCurrentlyUploading.set(false);
     }
 
-    private void processCloudWatchAttemptLogInformation(Map<String, Set<String>> completedLogFilePerComponent,
+    private void processCloudWatchAttemptLogInformation(Map<String, Set<LogFile>> completedLogFilePerComponent,
                                                         Map<String, CurrentProcessingFileInformation>
                                                                 currentProcessingLogFilePerComponent,
                                                         CloudWatchAttemptLogInformation attemptLogInformation,
-                                                        String fileName,
+                                                        String fileHash,
                                                         CloudWatchAttemptLogFileInformation
                                                                 cloudWatchAttemptLogFileInformation) {
-        //Todo: eventually the file and fileName should be obtained by the fileHash
-        LogFile file = new LogFile(fileName);
-        boolean isActiveFile = false;
-        //TODO: setting this flag is only to develop incrementally without having to changed all tests yet, so that
-        // we can avoid a massive PR. This will be removed in the end.
-        if (ACTIVE_LOG_FILE_FEATURE_ENABLED_FLAG.get()) {
-            try {
-                String fileHash = cloudWatchAttemptLogFileInformation.getFileHash();
-                LogFileGroup logFileGroup = attemptLogInformation.getLogFileGroup().syncDirectory();
-                Optional<LogFile> file1 = logFileGroup.getFile(fileHash);
-                // TODO: this is temporary for not changing the original logic, will be well handled.
-                if (!file1.isPresent()) {
-                    logger.atDebug().log("The fileHash does not exist in directory");
-                    return;
-                }
-                file = file1.get();
-                fileName = file.getAbsolutePath();
-                // TODO: the following logic is only for passing this small PR while not changing the current context.
-                //  We will grab the fileHash in the future.
-                Optional<LogFile> activeFile = logFileGroup.getActiveFile();
-                if (activeFile.isPresent()) {
-                    isActiveFile = activeFile.get().fileEquals(file);
-                }
-            } catch (InvalidLogGroupException e) {
-                logger.atDebug().cause(e).log();
+        try {
+            LogFileGroup logFileGroup = attemptLogInformation.getLogFileGroup().syncDirectory();
+            if (!logFileGroup.isHashExist(fileHash)) {
+                logger.atTrace().kv("fileHash", fileHash).log("component",
+                        logFileGroup.getFilePattern(), "File not found in directory");
```

Comment:
When would we reach this log? A file being deleted?

Reply:
when the file is under classification of processing or completed.